nf_deployments/v1alpha1/nf_deployed.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NFDeployedConnectivity struct {
+	// peer NF's Id (see NFInstance struct below)
+	NeighborName string `json:"neighborName,omitempty" yaml:"neighborName,omitempty"`
+}
+
+// NFDeployedInstance defines an NF instance that is deployed
+type NFDeployedInstance struct {
+	// unique ID for this NF instance
+	Id string `json:"id,omitempty" yaml:"id,omitempty"`
+	// name of workload cluster where the NF instance is to be deployed
+	ClusterName string `json:"clustername,omitempty" yaml:"clustername,omitempty"`
+	// type of NF, example: amf, smf, upf
+	NFType string `json:"nftype,omitempty" yaml:"nftype,omitempty"`
+	// flavor of NF, example: small, medium, large
+	NFFlavor string `json:"nfflavor,omitempty" yaml:"nfflavor,omitempty"`
+	// NF vendor name
+	NFVendor string `json:"nfvendor,omitempty" yaml:"nfvendor,omitempty"`
+	// the software version of this NF vendor's NFType
+	NFVersion string `json:"nfversion,omitempty" yaml:"nfversion,omitempty"`
+	// list of connected NF instances to this NF instance
+	Connectivities []NFDeployedConnectivity `json:"connectivities,omitempty" yaml:"connectivities,omitempty"`
+}
+
+// NFDeployedSpec defines list of deployed NF for Edge Status Aggregator to monitor
+type NFDeployedSpec struct {
+	NFInstances []NFDeployedInstance `json:"nfinstances,omitempty" yaml:"nfinstances,omitempty"`
```

Comment:
Why is this not part of the status of the topology?

Reply:
See the other comments. This is meant to drive ESA to start monitoring. As ESA is undergoing very little changes, we are keeping the CRD to start status collection the same as seed.

In a more conceptual view. The NFTopology CRD is meant to specify that for a cluster matching a label, a NFTemplate needs to be deployed onto that cluster, and the status fields there are meant to track the number of NF instances expected to be created as part of this deployment, and whether they are "deployed".

ESA CRD (which is what NFDeployed is meant for) is supposed to get a list of deployment that has happened (or expected to happen), and therefore it would start collecting statuses from the target cluster where this NF is expected to be deployed.

Reply:
thx for clarification. Would we not reflect the result in the topology CR?

Reply:
Meaning the (aggregation of) AMF/SMF/UPFDeployment statuses should be reflected on NFTopology CR? Probably not --- based on the CRD, there are basically two stages: the issuer of NFTopology has no prior knowledge of which NF will be deployed on what cluster, and therefore the controller for this CRD should track which NF is supposed to be deploying on which cluster. Typically, without aggregation logic, the expectation is for the user to read the status of NFTopology, and kubectl to each of the target workload cluster and check each deployed NF's status.

Reply:
ok here is my simplistic view.
Is topology not the overall controller?
if so I assume it drives nddeploy, not?
Is so nf deploy should now the origin of the stats and as such nf topology can get the aggregate back even w/o knowing how many got deployed.

Maybe too simplistic as I have not looked at the implementation, but this is what I would do.